pkg/capture/capture.go

```diff
@@ -396,18 +424,24 @@ func (c *Capture) lock() {
 func (c *Capture) unlock() {
 
 	// Signal that the rotation is complete, releasing the processing routine
+	// Since the done channel has a depth of one an Unblock() event needs to be
+	// sent to ensure that a capture currently waiting for packets in the buffering
+	// state continues to the next iteration in order to observe the unlock request
 	c.capLock.done <- struct{}{}
+	if err := c.captureHandle.Unblock(); err != nil {
```

Comment:
This little one actually gave me some headache (and took the longest to figure out, ironically). I always ran into deadlocks at first due to the `done` channel now also having a depth of one _and_ me trying to fetch the buffer element at the beginning of `lock()`. Since it was possible that a concurrent `status()` call (man am I glad I added these concurrency sledgehammer tests) was already queuing up _while_ the capture was in buffer mode _and_ blocked in the `PPOLL()` it was possible that everything got stuck.
This additional `Unblock()` will basically ensure that in such scenarios the number of calls is "balanced" again and there's always an unblock event to save the day... :rofl: